lib/internal/process/esm_loader.js

```diff
@@ -56,8 +57,23 @@ async function initializeLoader() {
 
   const { getOptionValue } = require('internal/options');
   const customLoaders = getOptionValue('--experimental-loader');
+  const preloadModules = getOptionValue('--import');
+  const loaders = await loadModulesInIsolation(customLoaders);
 
-  if (customLoaders.length === 0) return;
+  // Hooks must then be added to external/public loader
+  // (so they're triggered in userland)
+  esmLoader.addCustomLoaders(loaders);
+
+  // Preload after loaders are added so they can be used
+  if (preloadModules?.length) {
+    await loadModulesInIsolation(preloadModules, loaders);
+  }
+
+  isESMInitialized = true;
+}
+
+async function loadModulesInIsolation(specifiers, loaders = []) {
```

Comment:
There's no async op happening in here anymore (now that `ESMLoader::addCustomLoaders()` is sync, which happened whilst this PR was open), so I think remove the `async` here?

```suggestion
function loadModulesInIsolation(specifiers, loaders = []) {
```